src/main/java/net/tropicraft/core/common/entity/projectile/ExplodingCoconutEntity.java

```diff
@@ -29,10 +32,17 @@ public ExplodingCoconutEntity(World world, LivingEntity thrower) {
 
     @Override
     protected void onImpact(RayTraceResult result) {
-        // TODO - why isn't this being called?
         if (!world.isRemote) {
-            world.createExplosion(this, getPosX(), getPosY(), getPosZ(), 2.4F, Explosion.Mode.DESTROY);
-            remove();
```

Comment:
It looks like the rest of the logic for preventing throwing coconut bombs is in `ExplodingCoconutItem`, which might need to be updated to match this? Or not sure if this should be moved over there entirely.
If this does stay here though, will this cause unauthorised coconut bombs to stick around in the world after impact, since `remove` is never called?

Reply:
Yeah all the checks should be happening in the item for sure!